pkg/policy: remove unused types from policy package

ianvernon opened the pull request:
Given that conntrack cleanup is no longer done upon policy updates from userspace, the L3L4Policy is no longer needed. Remove it for cleaner code, as well as `SecurityIDContexts`, `L4RuleContexts`, and `L7RuleContexts`. Refactor functions which use these types to remove references to them. This makes the code cleaner and easier to understand, which will help in refactoring the policy package in the future to separate the updating of the datapath from policy computation itself.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
pkg/policy: remove L3L4Policy field from Consumable

Given that conntrack cleanup is no longer done upon policy updates from userspace, this
field is no longer needed. Remove it for cleaner code.

Signed-off by: Ian Vernon <ian@cilium.io>

rlenglet reviewed (commented):
I love it!

raybejjani commented on pkg/endpoint/policy.go:
```diff
@@ -355,11 +295,7 @@ func (e *Endpoint) resolveL4Policy(owner Owner, repo *policy.Repository, c *poli
 // and a map matching which rules were successfully removed.
 // Must be called with Consumable mutex held.
 func (e *Endpoint) regenerateConsumable(owner Owner, labelsMap *identityPkg.IdentityCache,
-	repo *policy.Repository, c *policy.Consumable) (changed bool, rulesAdd, rulesRm policy.SecurityIDContexts, err error) {
```
The function comment isn't consistent anymore. Could you remove `and a map matching which rules were successfully added/modified;  and a map matching which rules were successfully removed.` ?

raybejjani reviewed (approved):
I'm probably not the best person to review, but I did give it a proper read. If anyone else wanted to review this, they should speak up. Otherwise, hopefully this unblocks an excellent PR.

rlenglet reviewed (approved):
Awesome!

Commit:
pkg/policy: remove SecurityIDContexts and associated types

These are no longer used since CT cleanup is not done when endpoint policy is
recalculated.

Signed-off by: Ian Vernon <ian@cilium.io>
